dev/tools/dartdoc.dart

```diff
@@ -416,13 +412,14 @@ void putRedirectInOldIndexLocation() {
 void writeSnippetsIndexFile() {
   final Directory snippetsDir = Directory(path.join(kPublishRoot, 'snippets'));
   if (snippetsDir.existsSync()) {
-    final Iterable<String> files = snippetsDir
+    final Iterable<File> files = snippetsDir
         .listSync()
         .whereType<File>()
-        .map((File file) => path.basename(file.path))
-        .where((String name) => path.extension(name) == '.dart')
-        .map((String name) => name.substring(0, name.length - '.dart'.length));
-    File('$kPublishRoot/snippets.txt').writeAsStringSync(files.join('\n'));
+        .where((File file) => path.extension(file.path) == '.json');
+        // Combine all the metadata into a single JSON array.
+        final Iterable<String> fileContents = files.map((File file) => file.readAsStringSync());
+      final String jsonArray = '[\n${fileContents.join(',\n')}\n]';
```

Comment:
It might be better to actually interpret the JSON, add the object to an array, and write it out with the JSON encoder. Then things like indentation could be correct, and it would be easier to switch the representation to something else or add index-wide data (creation date of the index, etc.) if we wanted to. This isn't performance critical code.